packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -374,48 +354,28 @@ export class IsvDebugBootstrapExecutor extends SfdxCommandletExecutor<{}> {
     );
     const packageInfos = this.parsePackageInstalledListJson(packagesJson);
 
-    // 6: fetch packages
-    await this.executeCommand(
-      this.buildRetrievePackagesSourceCommand(
-        response.data,
-        packageInfos.map(entry => entry.name)
-      ),
-      { cwd: projectPath },
-      cancellationTokenSource,
-      cancellationToken
-    );
+    // 5a: create directory where packages are to be retrieved
+    shell.mkdir('-p', projectInstalledPackagesPath); // .sfdx/tools/installed-packages
+    const packageNames = packageInfos.map(entry => entry.name);
 
-    // 7a: unzip downloaded packages into temp location
-    try {
-      const packagesTempPath = path.join(projectMetadataTempPath, 'packages');
-      shell.mkdir('-p', packagesTempPath);
-      shell.mkdir('-p', projectInstalledPackagesPath);
-      const zip = new AdmZip(
-        path.join(projectMetadataTempPath, 'unpackaged.zip')
-      );
-      zip.extractAllTo(packagesTempPath, true);
-    } catch (error) {
-      console.error(error);
-      channelService.appendLine(
-        nls.localize('error_extracting_packages', error.toString())
-      );
-      notificationService.showErrorMessage(
-        nls.localize('error_extracting_packages', error.toString())
+    // 5b: retrieve packages
+    // TODO: what if packageNames.length is 0?
+    for (let i = 0; i < (packageNames.length); i++) {
+      await this.executeCommand(
+        this.buildRetrievePackagesSourceCommand(
+          response.data,
+          packageNames[i]
+        ),
+        { cwd: projectPath },
+        cancellationTokenSource,
+        cancellationToken
```

Comment:
Running `sfdx project:retrieve:start --package-name <value> ...` _i_ no. of times for _i_ no. of packages. The way sfdxCommandBuilder is building commands, it's not accepting a string/ array of package names as argument for `.withFlag()`.
e.g. packageNames = ['package1', "package pack2"] 
If I pass `.withflag(--package-name, packageNames.join(" "))` it treats  '`package1 "package pack2"`' as name of one package.

Reply:
Yes! if I pass ` pkg1 --package-name pkg2..` as string value to `--package-name` in `.withFlag()` it will error out saying `pkg1 --package-name pkg2..` is not a package.

And adding dynamic number of `.withFlag()` in `sfdxCommandBuilder` is a bit tricky. `sfdxCommandBuilder` extends `CommandBuilder` and that will lead to touching much more code. so preferring running the `sfdx project:retrieve:start` in a for loop.

Earlier we were retrieving in zip format and converting individual package. Now we are just retrieving individual package. It doesn't impact the performance as well.

Reply:
Thank you for sharing.